bots/tests-scan

```diff
@@ -458,10 +360,10 @@ def cockpit_tasks(api, update, branch_contexts, repo, pull_data, pull_number, sh
             return False
 
         if trigger_externals():
-            for proj_repo in REPO_BRANCH_CONTEXT:
+            for proj_repo in testmap.projects():
                 if proj_repo == "cockpit-project/cockpit":
                     continue
-                for context in REPO_BRANCH_CONTEXT[proj_repo].get("master", []):
+                for context in testmap.tests_for_project_branch(proj_repo, "master"):
```

Comment:
And here, use something like `tests_for_project(proj_repo).get("master", {})`, to avoid crashing on projects which only have non-master branches.

This is still unclean, but it will go away eventually with widening our support for triggering non-master branches. I. e. at that point it should run defined tests for all branches. You can do that either right here, or we can do it in a follow-up.